include/boost/url/impl/const_string.ipp

```diff
@@ -128,69 +128,67 @@ const_string::
 const_string(
     result const& r) noexcept
     : string_view(
-        r.data, r.size)
-    , p_(r.p)
+        r.data, r.size),
+        data_(r.p)
 {
 }
 
 const_string::
 ~const_string()
 {
-    p_->release(size());
+    if (size() > sizeof(base*)) {
+        data_.p_->release(size());
+    }
 }
 
 const_string::
 const_string() noexcept
-    : string_view()
-    , p_([]
-    {
-        struct empty_string final : base
-        {
-            void
-            destroy(std::size_t) noexcept override
-            {
-            }
-        };
-
-        struct deleter
-        {
-            empty_string* p;
-
-            ~deleter()
-            {
-                p->release(0);
-            }
-        };
-
-        static empty_string cs{};
-        static deleter d{&cs};
-
-        ++cs.refs;
-
-        return &cs;
-    }())
+    : string_view(data_.buf_, 0)
 {
 }
 
 const_string::
 const_string(
     const_string const& other) noexcept
     : string_view(other)
-    , p_(other.p_)
 {
-    ++p_->refs;
+    if (other.size() > sizeof(base*)) {
+        data_.p_ = other.data_.p_;
+        ++data_.p_->refs;
+    } else {
+        std::memcpy(
+            data_.buf_,
+            other.data_.buf_,
+            other.size());
+        static_cast<string_view&>(*this) =
+            string_view(data_.buf_, other.size());
+    }
 }
 
 const_string&
 const_string::
 operator=(
     const_string const& other) noexcept
 {
-    ++other.p_->refs;
-    p_->release(size());
-    p_ = other.p_;
-    static_cast<string_view&>(
-        *this) = other;
+    if (other.size() > sizeof(base*)) {
```

Comment:
Consider adding this private function:
```
constexpr bool const_string::is_small() const noexcept;
```
and calling it instead

Reply:
That was my first instinct. But I thought it would be better to let you suggest what to do.